mitodl/mit-learn
# Cg/fix podcast cta

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/6264

### Description (What does it do?)
This PR adds an `audio_url` field to the `PodcastEpisode` model and explicitly fills it with `rss_data.enclosure["url"]` during ETL. The `url` property of the `LearningResource` associated with a `PodcastEpisode` now defaults to `rss_data.link.text` (aka `episode_link`) and falls back to `audio_url` if that's not available. The frontend was set up to waterfall in the order of `episode_link` > `audio_url` > `url`.

### How can this be tested?
 - If you have Posthog set up locally, enable the `lr_drawer_v2` flag
 - If you don't have Posthog set up locally, you can force `drawerV2` to be `true` in `LearningResourceDrawer.tsx`
 - Spin up this branch of `mit-learn`
 - Run `docker compose exec web ./manage.py backpopulate_podcast_data` and wait for it to finish
 - Go to the search page at http://localhost:8062/ and search for "J. Mijin Cha, "A Just Transition for All: Workers and Communities for a Carbon-Free Future" (MIT Press, 2024)"
 - Click on the search result to bring up the drawer
 - Click on "Listen to Podcast" and verify that it links directly to the MP3 file for the podcast episode
